backend/cfg/cfg_liveness.ml

```diff
@@ -45,58 +45,48 @@ struct
       across : Reg.Set.t
     }
 
+  let instruction { before; across = _ } ~can_raise ~exn
+      (instr : _ Cfg.instruction) =
+    let across = Reg.diff_set_array before instr.res in
+    let across =
+      if can_raise then Reg.Set.union across exn.before else across
+    in
+    let before = Reg.add_set_array across instr.arg in
+    { before; across }
+
   let basic : domain -> exn:domain -> Cfg.basic Cfg.instruction -> domain =
-   fun { before; across = _ } ~exn instr ->
-    match instr.desc with
-    | Op _ | Call _ ->
-      if Cfg.is_pure_basic instr.desc
-         && Reg.disjoint_set_array before instr.res
-         && (not (Proc.regs_are_volatile instr.arg))
-         && not (Proc.regs_are_volatile instr.res)
-      then { before; across = before }
-      else
-        let across = Reg.diff_set_array before instr.res in
-        let across =
-          if Cfg.can_raise_basic instr.desc && instr.stack_offset > 0
-          then Reg.Set.union across exn.before
-          else across
-        in
-        let before = Reg.add_set_array across instr.arg in
-        { before; across }
-    | Reloadretaddr ->
-      { before = Reg.diff_set_array before Proc.destroyed_at_reloadretaddr;
-        across = Reg.Set.empty
-      }
-    | Pushtrap _ -> { before; across = before }
-    | Poptrap -> { before; across = before }
-    | Prologue -> { before; across = before }
+   fun ({ before; across = _ } as domain) ~exn instr ->
```

Comment:
I think we should keep a pattern (actually
an or-pattern listing all constructors), so
what we are notified if/when another
constructor is added.